### PR TITLE
Blog App. Milestone: Add Form

### PR DESCRIPTION
**Pull request by CaptainBawa**
### On this branch:

- [ ] I created a method called current_user in the ApplicationController to make current user data available to all controllers.
- [ ] I created forms to perform the following functions:
     - Create a Post on behalf of the current_user (use the method you created in your ApplicationController).
     - Could you create a comment for the current_user (use the method you created in your ApplicationController)
 
- [ ] Users can add likes to Posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,14 +1,76 @@
...
+  def create_comment
+    @post = Post.find(params[:id])
+    @comment = current_user.comments.build(comment_params.merge(post: @post))
+    if @comment.save
+      @comment.update_post_comments_counter
+      redirect_to user_post_path(@post), notice: 'Comment added successfully.'
+    else
+      render :show
+    end
+  end
```
Comment (Abenezer-Tilahun) [inline app/controllers/posts_controller.rb]: - [ ] Kindly I recommend you to  create **comments controller** inside **# controllers/comments_controller.rb** that use the method that you created in your Application-Controller outside of the **posts_controller.rb.**

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,14 +1,76 @@
...
+  def create_like
+    @post = Post.find(params[:id])
+    @like = current_user.likes.build(like_params.merge(posts_id: @post.id))
+    if @like.save
+      @like.update_post_likes_counter
+      redirect_to user_post_path(@post), notice: 'Post liked successfully.'
+    else
+      render :show
+    end
+  end
```
Comment (Abenezer-Tilahun) [inline app/controllers/posts_controller.rb]: - [ ] Kindly I recommend you to create **Likes controller** inside **# controllers/Likes_controller.rb** that use the method that you created in your Application-Controller outside of the **posts_controller.rb**.

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -1,14 +1,76 @@
...
+  # the `params` object in a Ruby on Rails application.
+  def post_params
+    params.require(:post).permit(:title, :text, :author_id, :comments_counter, :likes_counter)
+  end
+
+  # The function `comment_params` is used to extract and permit the `text` parameter
+  # from the `comment` object in the `params` hash.
+  def comment_params
+    params.require(:comment).permit(:text)
+  end
```
Comment (Abenezer-Tilahun) [inline app/controllers/posts_controller.rb]: - [ ] Kindly I recommend you to include this private method  **comment_params**  inside **controllers/comments_controller.rb .**

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -1,14 +1,76 @@
...
+
+  # The function `like_params` is used to permit specific parameters for the
+  # `like` object in a Ruby on Rails application.
+  def like_params
+    params.require(:like).permit(:users_id, :posts_id)
   end
```
Comment (Abenezer-Tilahun) [inline app/controllers/posts_controller.rb]: - [ ]  Kindly I recommend you to include this private method **like_params** inside **controllers/Likes_controller.rb** .

**Review by Abenezer-Tilahun — CHANGES_REQUESTED** [review]
Hi @CaptainBawa 

Great job for implemented  the some of the Project Requirement , but there are some changes that need to be implemented in order to proceed to the next project 👍

## Highlights 👏🏻
✔️ No Linter errors
✔️ Correct Git-Flow
✔️ Good commit messages
✔️ Professional README

almost

## Required Changes ♻️
_Check the comments below the review._


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DuaneDave — APPROVED** [review]
Hi @CaptainBawa 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Controllers methods are set up correctly
✔️ Clean and well structured file system

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**